### DISCUSSION TITLE
Have it working again on latest Pi4B bullseye 64bit

### DISCUSSION THREAD

**Discussion by K7MDL2** [cariboulabs/cariboulite#152] [General]
I had the CaribouLite running last May in a couple of apps but I have been away from the radios most all summer until now.  

This week I updated the OS (RPi4B with 64bit Bullseye OS on a SSD, 8GB RAM), deleted the old CaribouLite repo on my RPi, cloned the latest main repository, sorted out some dependencies.  Then fixed a nagging error about libglfw3 not found for sdr++.  That was quite involved with adding and removing chains of dependencies.
 
Had several errors running the CaribouLite build, mostly around SMI and fifo memory, poured through the latest open issues, updated my /boot/config.txt files per https://github.com/cariboulabs/cariboulite/issues/135.  That seemed to be the major trick.  I ended up changing several settings and adding new ones.  Here are the relevant settings I am using now distilled from the thread.

arm_64bit=1
dtparam=i2c_vc=on
arm_boost=1

[all]

# enable smi for CaribouLite SDR
enable_uart=0
dtparam=i2c_arm=off
#dtoverlay=smi
#dtoverlay=smi-dev
dtparam=spi=off
dtoverlay=spi1-3cs

Now have a clean build again for both SoapySDR and sdrpp.  I built sdrpp from source. 

Trying to run sdrpp --server it ran but when I connected from a client (on another machine, or not) CPU went to 100%.  This worked fine back in May. The sdrpp client and OpenWebRx work fine with my HL2 radios.  On the first connection and start, I hear a small audio crackle.  After that moment only control commands work (change freq), no audio streaming. I watched the network resources, no packets flowing for I/Q.  CPU pegged.  Odd.  A warning shows up per below.

Tried GQRX on my Windows 11 desktop but I got a destination port 1 out of range error dialog but otherwise it sees the SoapySDR info OK but no streaming data.  Version 2.15.8 on Linux Mint box.  No errors in the SoapySDR Server output but there are these 2 warnings on the GQRX and CubicSDR and I think the same was seen in SDR++ logs.

[WARNING] StreamEndpoint resize socket buffer: set 43008 KiB, got 1024 KiB
...
[WARNING] StreamEndpoint resize socket buffer: set 43008 KiB, got 208 KiB

After another fresh reboot tried CubicSDR v0.2.6a install from last Spring.  After fiddling with the settings/sliders a bit it started working and was receiver a FM stereo station.  I added a SoapyRemote entry with my Pi4 IP address (left port blank) to the device list under the Remote section though that is not required.  CPU usage was low, 11-26% range.  It ran for some minutes while typing this but then stopped for unknown reason.  I hit stop/start in the CubicSDR menu and again, after fiddling, it worked again.
![image](https://github.com/cariboulabs/cariboulite/assets/64443004/7e58df4a-08ec-4f38-a6a3-9614db155c1c)

Here is 98.1 FM stereo playing connected though a 4-way RX splitter on my 80-10M EFHW antenna.
![image](https://github.com/cariboulabs/cariboulite/assets/64443004/ea4801f9-c400-413c-8476-cf40bc53e1d5)

This is running at about 270Mbps.  CPU is doing around 30%.   

In the past I had the CaribouLite running on OpenWebRX+.  For some reason it did not think Soapy is enabled or installed.  Openbwebrx+ on Linux Mint on NUC4 box talks to the Soapy server on the Pi OK however it does not stream data.   Something was up with my OpenwebRX install.   I do have Openwebrx auto starting and at one point I found 2 instances running.  It is also serving a pair of HL2 sdrs and works well with those.

Here is my SoapySDRUtil --find output for reference:
![image](https://github.com/cariboulabs/cariboulite/assets/64443004/b0b7e0b6-af28-40a5-861e-84f40c62728f)

As I circled around running various SDR apps I rebooted several times. Magically sdrpp V1.1.0 Oct 11 build client on Win11 started to work using sdrpp server mode.  Listening to FM station at 200K BW. Audio is a bit chopped.  AGC gain control did not change the small chop. R educed the sample rate to 1.33Mhz and the chop minimized.   enet rate around 30Mbps.   CPU 11%. Audio in CubicSDr earlier seemed perfect.
 
Here is the cariboulite_test_app output:

Select a function:
 [0]  Hard reset FPGA
 [1]  Soft reset FPGA
 [2]  Print board info and versions
 [3]  Program FPGA
 [4]  Perform a Self-Test
 [5]  FPGA Digital I/O
 [6]  FPGA RFFE control
 [7]  FPGA SMI fifo status
 [8]  Modem transmit CW signal
 [9]  Modem receive I/Q stream
 [10]  Synthesizer 85-4200 MHz
 [99]  Quit
    Choice:   2

=====================================
Board Information (HAT)
	Category name: hat
	Product name: CaribouLite RPI Hat
	Product ID: 0x0001, Numeric: 1
	Product Version: 0x0001, Numeric: 1
	Product UUID: 4d7830e7-28bd-4bf8-9f95-d65f9081a6f2, Numeric serial: 0xFA504D6C
	Vendor: CaribouLabs LTD
11-02 05:02:40.440 32727 32727 D AT86RF215_Main at86rf215_print_version@at86rf215.c:290 MODEM Version: AT86RF215IQ (without basebands), version: 03
	Product Type: CaribouLite FULL
Library Versions:
	(Major, Minor, Rev): (1, 2, 0)

Here is the SDR++ server output showing 1 SMI read error, this is where I hear a quick audio blip then no audio:

![image](https://github.com/cariboulabs/cariboulite/assets/64443004/45585d7e-f785-4194-ad28-a5671c50f92a)

Bottom line, it can be made to work though it requires some work to get the right combo of OS + app + Server + drivers built.   I have more work to do to sort out the above open issues, many may be client-side apps, but I can see it is possible this can work.  Not seeing any errors on the SoapySDRServer output.  I know the developer is working on some key driver rewrites.  Seems like the hardware side is OK, looking forward to the software catching up to the hardware capabilities.  

Mike K7MDL

**Comment by a-oxide** [comment]
I appreciate your post! It's been very difficult getting it functional at all.
